odbc/src/api/odbc_uri.rs

```diff
@@ -0,0 +1,191 @@
+use crate::errors::{ODBCError, Result};
```

Comment:
This gives us a useful interface that should continue to work no matter what we do with the actually parsing in the `new` associated function come SQL-990. Note that this is a rather strange interface where all the methods are destructive, but this suits all of our use cases and this is a crate level interface. By writing it this way we can avoid copying some strings. Not a big deal, ultimately, but why waste memory when this is pretty easy?